File: src/main/java/com/mpe85/grampa/input/StringBufferInputBuffer.java
package com.mpe85.grampa.input;

public class StringBufferInputBuffer extends AbstractCharSequenceInputBuffer {
	
	public StringBufferInputBuffer(final StringBuffer stringBuffer) {
		super(stringBuffer);
		this.stringBuffer = stringBuffer;
	}
	
	@Override
	public int getCodePoint(final int index) {
		return stringBuffer.codePointAt(index);
	}
	
	
	private final StringBuffer stringBuffer;
	
}

File: src/main/kotlin/com/github/mpe85/grampa/input/StringBufferInputBuffer.kt
package com.github.mpe85.grampa.input

/**
 * An [InputBuffer] implementation using a [StringBuffer].
 *
 * @property[stringBuffer] A string buffer backing the input buffer
 * @author mpe85
 */
public class StringBufferInputBuffer(private val stringBuffer: StringBuffer) :
    CharSequenceInputBuffer(stringBuffer) {

    override fun getCodePoint(index: Int): Int = stringBuffer.codePointAt(index)
}
